src/main/java/io/pravega/connectors/flink/FlinkPravegaReader.java

```diff
@@ -268,7 +269,10 @@ public void run(SourceContext<T> ctx) throws Exception {
             // main work loop, which this task is running
             while (this.running) {
                 final EventRead<T> eventRead = pravegaReader.readNextEvent(eventReadTimeout.toMilliseconds());
-                final T event = eventRead.getEvent();
+
+                final T event = this.deserializationSchema instanceof PravegaDeserializationSchema ?
```

Comment:
Consider doing the `instanceof` check once, outside the `while` loop, for better performance.   One approach would be like:
```
Function<EventRead,T> f = this.deserializationSchema instanceof PravegaDeserializationSchema ? this.deserializationSchema::deserializeWithMetadata : (eventRead) -> eventRead.getEvent();
```

Reply:
That's very helpful. I'll change it.